### PR TITLE
#781: Implement RadioGroup (closes #781)

### LINKED ISSUE #781: Implement RadioGroup
## requirements
This component should be able to render radio buttons in a completely custom layout. Each radio button should be correctly on/off based on the state passed to its parent (radiogroup).

See also https://zpl.io/a3qYxEy

## specs
{optional: other useful info}

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by giogonzo**
Closes #781

as a form field:
screenshot 2018-12-28 at 10 44 38

vertical:
screenshot 2018-12-28 at 10 44 24

horizontal:
screenshot 2018-12-28 at 10 44 32

### tests performed

> A Test Plan is used to show what you tested to make sure your code works fine. You should write here all that you did to test, and provide some results of your testing.

> These results can be screenshots, query results, or even just pointers to unit tests that successfully passed.

### tests not performed (domain coverage)

> At times not everything can be tested, and writing what hasn't been tested is just as important as writing what has been tested.

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.

### CODE REVIEW

File: src/RadioGroup/RadioGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  value: string;
+};
+
+export type RadioGroupRequiredProps = {
+  /** name for the group */
+  name: string;
+  /** value */
+  value?: string;
+  /** onChange */
+  onChange: (value: string) => void;
```
Comment (FrancescoCioria) [inline src/RadioGroup/RadioGroup.tsx]: is it the same as `Dropdown`? are we using the string value instead of the option?

**Comment by FrancescoCioria** [comment]
@giogonzo I think we'd also like to use them stacked vertically instead of horizontally, is this possible with your code?

File: src/RadioGroup/RadioGroup.tsx (hunk 1 above)
Comment (giogonzo) [reply src/RadioGroup/RadioGroup.tsx]: `Dropdown` is more complex: `export type Value = SelectNS.Option | SelectNS.Options | SimpleValue;`

I kept things as simple as possible here (only "simple value", and harcoded to `string`). Do you think we should add more flexibility?

**Comment by giogonzo** [comment]
> I think we'd also like to use them stacked vertically instead of horizontally, is this possible with your code?

@FrancescoCioria I'll give it a try, should we default to horizontal or vertical?

File: src/RadioGroup/RadioGroup.tsx (hunk 1 above)
Comment (FrancescoCioria) [reply src/RadioGroup/RadioGroup.tsx]: no, it's enough for now (also this is controversial in `Dropdown` as well...)

**Comment by FrancescoCioria** [comment]
> @FrancescoCioria I'll give it a try, should we default to horizontal or vertical?

discussed with alessandra, the default should be vertical

File: src/RadioGroup/radioGroup.scss [hunk 2]
```diff
@@ -69,4 +68,12 @@ $opacity-disabled: 0.6 !default;
       cursor: auto;
     }
   }
+
+  &:not(.is-horizontal) .radio-group-option:not(:last-child) {
+    margin-bottom: 8px;
+  }
+
+  &.is-horizontal .radio-group-option {
```
Comment (FrancescoCioria) [inline src/RadioGroup/radioGroup.scss]: missing `:not(:last-child)`?

**Pull request merged**